### PR TITLE
aixPb: Add steps to install dnf and deprecate yum for AIX 7.2 and higher

### PR DESCRIPTION
**Pull request by Haroon-Khel**
- [x] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [ ] [faq.md](https://github.com/adoptium/infrastructure/blob/master/FAQ.md) updated if appropriate
- [ ] other documentation is changed or added (if applicable)
- [ ] playbook changes run through [VPC](https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptium.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)
- [x] VPC/QPC not applicable for this PR
- [ ] for inventory.yml changes, bastillion/nagios/jenkins updated accordingly

ref https://github.com/adoptium/infrastructure/issues/3142#issuecomment-1662156865

### CODE REVIEW

**Comment by Haroon-Khel** [comment]
As per @aixtools's comments in https://github.com/adoptium/infrastructure/issues/3149#issuecomment-1671000103 I am also in favour of using dnf only and deprecating yum

**Comment by Haroon-Khel** [comment]
@aixtools I am not too familiar with https://github.com/adoptium/infrastructure/blob/1e3dfa0386893a1b023852d3c1b41be7e2a1d1aa/ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/yum/tasks/main.yml#L15 

Do you think it needs to be included in the dnf install?

**Review by sxa — CHANGES_REQUESTED** [review]
These removals make me shudder a little, but if we can add a checksum into the toolbox download that would be good t avoid potential tampering ... Not sure how often that script gets updated though, so we may have to keep updating it.

File: ansible/playbooks/AdoptOpenJDK_AIX_Playbook/main.yml [hunk 1]
```diff
@@ -64,7 +64,8 @@
     # Note: AIX File system configuration must be run now as RPM based software
     # cannot expand filesystem space on demand
     - aixfs
-    - yum
+    - role: dnf
+      tags: dnf
```
[the remark this answers is not in the document]
Comment (Haroon-Khel) [reply ansible/playbooks/AdoptOpenJDK_AIX_Playbook/main.yml:68]: There might be times where we want to run only this role

**Review by sxa — APPROVED** [review]
LGTM

**Comment by Haroon-Khel** [comment]
Going to test this on https://ci.adoptium.net/computer/test%2Dosuosl%2Daix72%2Dppc64%2D5/, its yum will be upgraded to dnf

**Comment by Haroon-Khel** [comment]
Installed openssl3 on test-osuosl-aix72-ppc64-5. Doesnt look like there was a problem

```
root@adopt05:[/root/openssl3/openssl-3.0.10.1000]openssl version
OpenSSL 3.0.10 1 Aug 2023 (Library: OpenSSL 3.0.10 1 Aug 2023)
```

**Comment by Haroon-Khel** [comment]
Its having trouble installing ca-certificates-2023.2.60-0.aix7.1.ppc.rpm which is included in the dnf bundle. A lot of conflicting files between ca-certificates-2023.2.60-0.aix7.1.ppc.rpm and the ca certs thats already installed, ca-certificates-2021.2.52-1.ppc. For example

```
file /opt/freeware/bin/update-ca-bundles from install of ca-certificates-2023.2.60-0.ppc conflicts with file from package ca-certificates-2021.2.52-1.ppc
        file /opt/freeware/etc/ssl/certs/ACCVRAIZ1.crt from install of ca-certificates-2023.2.60-0.ppc conflicts with file from package ca-certificates-2021.2.52-1.ppc
        file /var/ssl/certs/ACCVRAIZ1.crt from install of ca-certificates-2023.2.60-0.ppc conflicts with file from package ca-certificates-2021.2.52-1.ppc
        file /opt/freeware/etc/ssl/certs/AC_RAIZ_FNMT-RCM.crt from install of ca-certificates-2023.2.60-0.ppc conflicts with file from package ca-certificates-2021.2.52-1.ppc
        file /var/ssl/certs/AC_RAIZ_FNMT-RCM.crt from install of ca-certificates-2023.2.60-0.ppc conflicts with file from package ca-certificates-2021.2.52-1.ppc
        file /opt/freeware/etc/ssl/certs/AC_RAIZ_FNMT-RCM_SERVIDORES_SEGUROS.crt from install of ca-certificates-2023.2.60-0.ppc conflicts with file from package ca-certificates-2021.2.52-1.ppc
        file /var/ssl/certs/AC_RAIZ_FNMT-RCM_SERVIDORES_SEGUROS.crt from install of ca-certificates-2023.2.60-0.ppc conflicts with file from package ca-certificates-2021.2.52-1.ppc
        file /opt/freeware/etc/ssl/certs/ANF_Secure_Server_Root_CA.crt from install of ca-certificates-2023.2.60-0.ppc conflicts with file from package ca-certificates-2021.2.52-1.ppc
        file /var/ssl/certs/ANF_Secure_Server_Root_CA.crt from install of ca-certificates-2023.2.60-0.ppc conflicts with file from package ca-certificates-2021.2.52-1.ppc
```
(without posting the whole log)

**Comment by Haroon-Khel** [comment]
Updating ca-certificates to ca-certificates-2021.2.52-3 I was able to fix the above error. Dnf successfully installed

**Comment by Haroon-Khel** [comment]
Need to specify a python 3 interpreter 

```
failed: [test-osuosl-aix72-ppc64-5] (item=bash) => {"ansible_loop_var": "item", "changed": false, "item": "bash", "msg": 
"Could not import the dnf python module using /usr/bin/python (2.7.18 (default, Dec 15 2022, 05:12:43) [GCC 8.3.0]). Please
 install `python3-dnf` or `python2-dnf` package or ensure you have specified the correct ansible_python_interpreter. 
(attempted ['/usr/libexec/platform-python', '/usr/bin/python3', '/usr/bin/python2', '/usr/bin/python'])", "results": []}
```

**Comment by Haroon-Khel** [comment]
Changing the python interpreter to `ansible_python_interpreter: /opt/freeware/libexec/python3` for whenever the `dnf` module is used works

more info here https://community.ibm.com/community/user/power/discussion/dnf-works-but-unable-to-import-dnf-python-module

**Comment by Haroon-Khel** [comment]
Same problem with the cmake install

```
TASK [dnf : Install cmake 3.14.3 (See https://github.com/AdoptOpenJDK/openjdk-build/issues/2492)] ***
fatal: [test-osuosl-aix72-ppc64-5]: FAILED! => {"changed": false, "failures": ["cmake-3.14.3 All matches were filtered out by exclude filtering for argument: cmake-3.14.3"], "msg": "Failed to install some of the specified packages", "rc": 1, "results": []}
```
Should be made more idempotent

**Comment by Haroon-Khel** [comment]
> Lets see what adding a disable_excludes: all does. It should still install the excluded packages if theyre not present, and do nothing if theyre present since in the install list the autoconf and freetype2-devel include their version numbers

Added this to the cmake install. Should be fine if theres no cmake-3.14.3 present. Should do nothing if it is present

**Comment by Haroon-Khel** [comment]
https://awx2.adoptopenjdk.net/#/jobs/playbook/2256?job_search=page_size:20;order_by:-finished;not__launch_type:sync

Looks good

**Comment by Haroon-Khel** [comment]
Id say this is good to merge

**Pull request merged**